File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-5.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        
        stack = []

        for num in tokens:
            match num:
                case "+":
                    b = stack.pop()
                    a = stack.pop()
                    res = int(a) + int(b)
                    stack.append(str(res))
                
                case "-":
                    b = stack.pop()
                    a = stack.pop()
                    res = int(a) - int(b)
                    stack.append(str(res))
                
                case "*":
                    b = stack.pop()
                    a = stack.pop()
                    res = int(a) * int(b)
                    stack.append(str(res))
                
                case "/":
                    b = stack.pop()
                    a = stack.pop()
                    res = int(int(a) / int(b))
                    stack.append(str(res))
                
                case _:
                    stack.append(num)
            
        return int(stack[-1])

File: Data Structures & Algorithms/daily-temperatures/submission-5.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<>();
        int size = temperatures.length;

        int[] res = new int[size];

        for(int i = 0; i < size; i++){
            while(!stack.isEmpty() && temperatures[stack.peek()] < temperatures[i]){
                int stackIndex = stack.pop();
                res[stackIndex] = i - stackIndex;
                
            }
            stack.push(i);
        }

        return res;
    }
}
